### ISSUE TITLE
The playlist functionality got broken (probably after trying to add a non existing video)

### ISSUE THREAD

**Issue by meiphoo7-Mae** [libre-tube/LibreTube#2546]
### Steps to reproduce

I'm not sure about this. I assume the problem has something to do with trying to add a non existing url, but that might be a wrong assumption.

* Try to add a non existing video url to the playlist
The app shows a message about a server error and suggests to try a different instance.
* Then close the app.
* Start the app again and try to add an existing video to the playlist.
Now the playlist is still visible, but adding and deleting video's is not possible.

### Expected behavior

When trying to add an non existing video, it should be ignored.

### Actual behavior

When trying to add an non existing video the app shows a message about a server error and suggests to try a different instance. I use the official one so I didn't try a different one.
After this the app keeps working.
When the app is closed and then restarted. It is not longer possible to add a new video to the playlist or delete one from it, also the playlist can't be deleted and it is not possible to create a new playlist.
However despite the seemingly broken playlist it is still possible to play videos and they also appear in the watch history.
Also the settings can be exported and creating a backup is possible. Restoring the backup (over the current settings) does not solve the problem. I don't know if the backup is a good one because I can't judge if the json content is valid.

### LibreTube version

0.9.1

### Android version

Android 10

### Other details

I imported the non existing url from bookmarks in my browser and did not view it before trying to add it to the app.
After the error message in the app I did try to view the video in the browser and youtube showed that the channel and associated videos were deleted.

Additional question:
Can I somehow restore the playlist to a working state?

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
How did you add a non existing video to the playlist? You can only add videos to a playlist that you can access within the app, or did you do it by sharing the link to the app?
Are you signed in with an account?

**Comment by meiphoo7-Mae** [comment]
Yes, I shared a browser bookmark with the app and I'm signed in with an account.

**Comment by Bnyro** [comment]
When it's been created while logged in with an account it's only saved to that specific account but not inside the app, so I don't think there's anything we can do here (at LibreTube's side).
I assume it doesn't work either on the official frontend of the instance you're using, e.g. <https://piped.kavin.rocks>?
If it doesn't it might be the best idea to contact the owner of the Piped instance you're using and ask them for help, they could maybe restore it from the database.

**Comment by meiphoo7-Mae** [comment]
I don't understand that login stuff at all. Currently I can logout, but it says it is only for a LibreTube account. I can also delete an account but then I delete a Piped account?!
So two different accounts while I've logged it only once.

**Comment by Bnyro** [comment]
When logged in with a account (which always is from Piped), playlists and subscriptions get stored at the Piped servers. When not being logged in, everything is only stored on your phone. So when you delete an account, you delete a Piped account, yes.

> And yes: the instance is kavin.rocks.
>
You mean the playlist is visible and working at <https://piped.kavin.rocks/playlists> when logged in with your account?

**Comment by meiphoo7-Mae** [comment]
Sorry, I misread your previous question. I thought you were asking about the instance I used, that why I mentioned that it was kavin.rocks. I now understand that you asked about what happens when I'm logged in.

When logged in (at kavin.rocks) it shows the symptoms mentioned in the first post. When logged out (and the playlist restored from a backup) everything is working fine. I can again add and delete videos so the local handling is fine.
However now there's new problem. When I try to log in again, it seems to do nothing and when I push the "back" button of Android to remove the login screen it flashes a few times the message:

Undispatched coroutine was cancelled

I can imagine what the "cancelled" part means because pushing the "back" button is in fact cancelling. I don't know what "Undispatched coroutine" means.

I didn't try yet what happens when I create a new account and I don't know yet if I'm going to do that as I don't understand what the added benefit is of a Piped account because without a Piped account the app seems to work just fine.

**Comment by meiphoo7-Mae** [comment]
> I don't understand that login stuff at all. Currently I can logout, but it says it is only for a LibreTube account. I can also delete an account but then I delete a Piped account?! So two different accounts while I've logged it only once.

Concerning the above quote:
It is not what I thought: Both logout (or as a matter of fact Login/Register) AND deleting an account is about a Piped account according to the English translation. The translation for my language states otherwise but is wrong and there are more sentences incorrectly translated for my language so I switched to English now.

**Comment by Bnyro** [comment]
1. I actually wanted to you to test how it behaves when trying to access in the browser via <https://piped.kavin.rocks/playlists>, you already described before that it doesn't inside LibreTube. So I would kindly ask to check that now :)
2. The benefit of a Piped account is that it's in sync across multiple devices, which means you can use it on the browser and LibreTube on different devices while keeping the same subscriptions and playlists.

**Comment by meiphoo7-Mae** [comment]
Thanks for clarifying.
I didn't know I could access the playlist in a browser.

Now it is getting a little weirder. In a browser I can actually add a video to the playlist, delete it again and I can rename the playlist. I did not try if I can delete it. And I can create a new empty playlist (which I can delete).

So now the situation is:
1) Without an account I can do everything I want to do in LibreTube.
2) In a browser I can manage my playlists on Piped.

So apparently something is off in the interaction between LibreTube and Piped.
Can the 'export to JSON' function of Piped probably fix things?

I've also tried to delete LibreTube and reinstall it but that doesn't make a difference.

**Comment by meiphoo7-Mae** [comment]
>  If I got you correctly, when logged in with the account, everything works fine on the web with Piped, but when doing the same in LibreTube it doesn't?

Yes, that **_was_** correct.

The situation is now as follows:
In LibreTube I tried to delete my local playlist in preparation of testing the export from Piped. It didn't work. Then I tried to delete it again (I don't like to get defeated by devices and/or software... :grin:).
A popup appeared with messages about java exceptions, but the playlist got deleted anyway. Then I logged in and everything from my Piped account was imported.
I tried to create and delete a new playlist in LibreTube (it worked). I Added and deleted a random video to the existing/imported playlist and that worked too. So after all I didn't try the export from Piped because LibreTube imported everything automatically.

So it seems the problem somehow disappeared. I guess it must be caused by something stuck on the cache partition of Android that seemed to have survived even when the app was deleted and reinstalled.

I didn't save the error messages because I've seen Java related error messages before when using a different app and the developer told me the messages were too generic.

Well, this problem seems to be solved for now. Unfortunately we haven't been able to identify a probable cause.
Thanks for your assistance and if I can provide help for something the project may benefit from please let me know.

**Comment by Bnyro** [comment]
As long as it's working now we're all fine, the thing with the crash when deleting in LibreTube is a known issue and will be fixed in the next update. Maybe the issue has also been caused by some problems with the database of the instance you were using, who knows.
Anyways, going to close the issue for now!
